### DISCUSSION TITLE
Trouble creating new filter

### DISCUSSION THREAD

**Discussion by paulm17** [fabricjs/fabric.js#8132] [Q&A] [answered]
I'm wanting to create my own custom non-webgl filter.

Read: http://fabricjs.com/fabric-filters

Found SO questions like: https://stackoverflow.com/questions/36082793/how-to-create-a-custom-filter-by-passing-a-color-code-in-fabric-js

But seems those examples are too old, perhaps?

I've created a sandbox: https://codesandbox.io/s/beautiful-easley-pdn52d?file=/src/App.js

I get as an error: 

> canvasEl.getContext is not a function

If someone passing by can fix this for v5, I'd appreciate it!  I just need a working sample to then take it from there!  😄

**Comment by ShaMan123** [comment, marked as answer]
The signature used in SO is wrong or outdated (luckily TS is on it's way)
The [link](http://fabricjs.com/fabric-filters) you provided shows it clearly

First diable webgl backend:
```diff

+ fabric.enableGLFiltering = false;
```

Then look at `applyTo`, you have a lot of context in that method as the first arg (not a canvas el)

https://codesandbox.io/s/trusting-benji-xmewt4?file=/src/App.js

**Reply by paulm17** [reply]
Thank you so much for taking the time to answer and provide a working example.  Really appreciate it!
